plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -451,9 +603,46 @@ public function get_description(): string {
 	 * @param WP_Post $post The deleted post.
 	 */
 	private function handle_deleted_post( $postid, $post ): void {
-		if ( 'shop_order' === $post->post_type && $this->data_sync_is_enabled() ) {
+		global $wpdb;
+
+		if ( 'shop_order' !== $post->post_type && 'shop_order_refund' !== $post->post_type ) {
+			return;
+		}
+
+		$features_controller = wc_get_container()->get( FeaturesController::class );
```

Comment:
this should check if custom order tables exist, instead of the feature being enabled, we will enable the feature anyway for everyone with WC 8.0 (although we won't set HPOS as authoritative).

Reply:
But will the user still have the ability the turn off the feature?

If yes: we still need to check the feature anyway.
If not: do we need any check at all? Can't we just assume that the tables exist at this point?

Asking because I suspect that checking the feature is faster than checking if the table exists.

Reply:
> I suspect that checking the feature is faster than checking if the table exists.

Makes sense, checking for tables will indeed be slow. I can link to this feature being enabled with tables being created, so we can keep this check as it is in that case.